packages/runtime/src/widgets/Date.tsx

```diff
@@ -0,0 +1,361 @@
+import {
+  type EnsembleAction,
+  useRegisterBindings,
+} from "@ensembleui/react-framework";
+import { Form as AntForm, Typography } from "antd";
+import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
+import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
+import { DatePicker } from "@mui/x-date-pickers/DatePicker";
+import { useState, createContext, useContext, useCallback } from "react";
+import {
+  PickersCalendarHeader,
+  type PickersCalendarHeaderProps,
+} from "@mui/x-date-pickers/PickersCalendarHeader";
+import {
+  PickersActionBar,
+  type PickersActionBarProps,
+} from "@mui/x-date-pickers/PickersActionBar";
+import { Button, IconButton, TextField } from "@mui/material";
+import dayjs from "dayjs";
+import CalendarTodayIcon from "@mui/icons-material/CalendarToday";
+import { Edit } from "@mui/icons-material";
+import { toString } from "lodash";
+import { WidgetRegistry } from "../registry";
+import type { EnsembleWidgetProps } from "../shared/types";
+import { useEnsembleAction } from "../runtime/hooks/useEnsembleAction";
+import type { FormInputProps } from "./Form/types";
+
+const DateHeaderFormat = "ddd, MMM DD";
+const DateDisplayFormat = "MM/DD/YYYY";
+
+type DateProps = {
+  initialValue?: string;
+  firstDate?: string;
+  lastDate?: string;
+  showCalendarIcon?: boolean;
+  onChange?: EnsembleAction;
+} & EnsembleWidgetProps &
+  FormInputProps<string>;
+
+interface DatePickerProps {
+  firstDate?: string;
+  lastDate?: string;
+  value?: dayjs.Dayjs | undefined;
+  setValue?: (value: dayjs.Dayjs | undefined) => void;
+  isCalendarOpen?: boolean;
+  setIsCalendarOpen?: (value: boolean) => void;
+  enteredDate?: string;
+  setEnteredDate?: (value: string) => void;
+  errorText?: string;
+  setErrorText?: (value: string) => void;
+}
+
+export const DatePickerContext = createContext<DatePickerProps | undefined>(
+  undefined,
+);
+
+export const Date: React.FC<DateProps> = (props) => {
+  const [value, setValue] = useState<dayjs.Dayjs | undefined>(
+    props.initialValue ? dayjs(props.initialValue) : undefined,
+  );
+  const [openPicker, setOpenPicker] = useState(false);
+  const [isCalendarOpen, setIsCalendarOpen] = useState(false);
+  const [enteredDate, setEnteredDate] = useState("");
+  const [errorText, setErrorText] = useState("");
+
+  const { values } = useRegisterBindings({ ...props, value }, props.id, {
+    setValue,
+  });
+
+  const action = useEnsembleAction(props.onChange);
+  const onChangeCallback = useCallback(
+    (date: any) => {
+      const formattedDate = dayjs(toString(date))?.format(DateDisplayFormat);
+      formattedDate !== "Invalid Date" && setEnteredDate(formattedDate);
```

Comment:
This should be a separate hook to sync the date with state. Hooks should be responsible for one thing